follower/follower_builder.go

```diff
@@ -0,0 +1,745 @@
+	followereng "github.com/onflow/flow-go/engine/common/follower"
+	synceng "github.com/onflow/flow-go/engine/common/synchronization"
+	"github.com/onflow/flow-go/model/encodable"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/model/flow/filter"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/buffer"
+	finalizer "github.com/onflow/flow-go/module/finalizer/consensus"
+	"github.com/onflow/flow-go/module/id"
+	"github.com/onflow/flow-go/module/mempool/stdmap"
+	"github.com/onflow/flow-go/module/metrics"
+	"github.com/onflow/flow-go/module/synchronization"
+	"github.com/onflow/flow-go/network"
+	netcache "github.com/onflow/flow-go/network/cache"
+	cborcodec "github.com/onflow/flow-go/network/codec/cbor"
+	"github.com/onflow/flow-go/network/p2p"
+	"github.com/onflow/flow-go/network/validator"
+	"github.com/onflow/flow-go/state/protocol"
+	badgerState "github.com/onflow/flow-go/state/protocol/badger"
+	"github.com/onflow/flow-go/state/protocol/blocktimer"
+	storage "github.com/onflow/flow-go/storage/badger"
+)
+
+// ObserverBuilder extends cmd.NodeBuilder and declares additional functions needed to bootstrap an Access node
+// These functions are shared by staked and observer builders.
+// The Staked network allows the staked nodes to communicate among themselves, while the public network allows the
+// observers and an Access node to communicate.
+//
+//                                 public network                           staked network
+//  +------------------------+
+//  | observer 1             |<--------------------------|
+//  +------------------------+                           v
+//  +------------------------+                         +----------------------+              +------------------------+
+//  | observer 2             |<----------------------->| Access Node (staked) |<------------>| All other staked Nodes |
+//  +------------------------+                         +----------------------+              +------------------------+
+//  +------------------------+                           ^
+//  | observer 3             |<--------------------------|
+//  +------------------------+
+
+type ObserverBuilder interface {
+	cmd.NodeBuilder
+}
+
+// ObserverServiceConfig defines all the user defined parameters required to bootstrap an access node
+// For a node running as a standalone process, the config fields will be populated from the command line params,
+// while for a node running as a library, the config fields are expected to be initialized by the caller.
+type ObserverServiceConfig struct {
+	bootstrapNodeAddresses  []string
+	bootstrapNodePublicKeys []string
+	bootstrapIdentities     flow.IdentityList // the identity list of bootstrap peers the node uses to discover other nodes
+	NetworkKey              crypto.PrivateKey // the networking key passed in by the caller when being used as a library
+	baseOptions             []cmd.Option
+}
+
+type PublicNetworkConfig struct {
```

Comment:
I don't think this is used anymore

Reply:
Fixed: https://github.com/GetElastech/flow-go/commit/b558c135a1c6516b298a383c819e841b8c4106d1

Reply:
Fixed: https://github.com/GetElastech/flow-go/commit/9f2e4956b1d7724deaa184a2157b94d8b23bc618